master/buildbot/steps/source/p4.py

```diff
@@ -150,11 +151,11 @@ def full(self, _):
         if self.revision:
             if debug_logging:
                 log.msg("P4: full() sync command based on :base:%s changeset:%d", self._getP4BaseForLog(), int(self.revision))
-            yield self._dovccmd(['sync', '%s...@%d' % (self._getP4BaseForCommand(), int(self.revision))], collectStdout=True)
+            yield self._sync(['%s...@%d' % (self._getP4BaseForCommand(), int(self.revision))])
```

Comment:
Perforce’s `#rev` is a per-file revision number, whereas `@change` is a global, per-commit revision number (also called changeset or changelist). It’s inherited from CVS. Since the code syncs whole directories, the `@` notation is correct here.